brlaser: switch to more-maintained upstream, same as Fedora

ahydronous opened the pull request:
Fedora switched to a more-maintained upstream fork because it supports more printers.

See: https://packages.fedoraproject.org/pkgs/printer-driver-brlaser/printer-driver-brlaser/index.html and
https://github.com/pdewacht/brlaser/pull/92

## Description of changes
Changed upstream from https://github.com/pdewacht/brlaser to https://github.com/Owl-Maintain/brlaser".
<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

ahydronous committed:
brlaser: switch to same more-maintained upstream as Fedora

brlaser: Fedora switched to a more-maintained upstream fork because it supports more printers.

See: https://packages.fedoraproject.org/pkgs/printer-driver-brlaser/printer-driver-brlaser/index.html
and
https://github.com/pdewacht/brlaser/pull/92

brlaser: switch to same more-maintained upstream as Fedora

Fedora switched to a more-maintained upstream fork because it supports more printers.

See: https://packages.fedoraproject.org/pkgs/printer-driver-brlaser/printer-driver-brlaser/index.html
and
https://github.com/pdewacht/brlaser/pull/92

winterqt reviewed (changes requested):
Please rebase on top of master and switch the target branch to it, and change your commit message to include the attribute at the front (`brlaser: ...`).

ahydronous commented:
I'll do that tomorrow!

I was looking for something tangentially related and found these two issues:
https://github.com/NixOS/nixpkgs/issues/291161
https://github.com/NixOS/nixpkgs/issues/290431

I'm linking them here so they can be closed if the PR is accepted.

ahydronous committed:
brlaser: refer to supported printer list in more concise way due to huge list.

brlaser: newer `brlaser` upstream supports a big chunk of extra printers, making the list too large to fit onto the package info page.
Refering to the supported printers in this way also means that no edits are necessary when new printer support is added.

ahydronous commented:
@winterqt I cherrypicked the commit(s) and rebased them on master. But I can only switch the target branch, not my head branch. Should I just open a new pull request? Or am I misunderstanding something.

winterqt commented:
1. You didn't actually push the rebased commits. (Technically you don't need to, but could you do that? You'll need to pass `--force`.)
2. You did the branch change correctly!

ahydronous commented:
Hopefully this is correct?
I did add a second commit, because I realized that I would have to add the extra supported printers in the list, and when I realized that the list would be way too long, I decided to refer to the supported printers in this way instead. Hope that's ok!

Stekke reviewed (approved):
pdewacht no longer has access to a brother laser printer, so development by him has stopped. The fork seems good.

Stekke commented:
Actually @ahydronous would you like to take over maintenance of this package from me? I don't have a brother laser printer anymore either.

ahydronous commented:
@Stekke yea no problem! I'm guessing that's something that should be put into a separate PR?

winterqt commented:
No, this can be done here. Just make a separate commit for it.

ahydronous committed:
maintainer-list.nix: add ahydronous to maintainer list

ahydronous committed:
brlaser: "ahydronous" replaces "stijnDW" as maintainer

ahydronous commented:
@winterqt I've done the changes to the maintainer list and brlaser maintainer!

admiral-shlork commented:
hello there, any chance someone could move this along, please? looks like it just needs a rebase it, thanks a lot!

onny commented:
Would be happy to review this PR, probably needs a rebase :+1:

onny commented:
Merged with https://github.com/NixOS/nixpkgs/pull/380593

Thank you @ahydronous for the initiative
